ta/mk/build-user-ta.mk

```diff
@@ -2,18 +2,18 @@ include mk/cleanvars.mk
 sm := $(lastword $(subst /, ,$(dir $(ta-mk-file))))
 sm-$(sm) := y
 
+# Select TA target (aka TA dev kit, when delivered)
+ta-target := $(strip $(if $(CFG_USER_TA_TARGET_$(sm)), \
+		$(filter $(CFG_USER_TA_TARGET_$(sm)), $(ta-targets)), \
+		$(default-user-ta-target)))
+
 ifeq ($(ta-target),ta_arm32)
 arch-bits-$(sm) := 32
```

Comment:
You're checking `$(ta-target)`, but then you're using `$(sm)`. I can't make sense of that.

Reply:
`sm`, from _build-user-ta.mk_ is the in-tree TA name (avb, pkcs11, ...).
As build process uses `CFLAGS$(arch-bits-$(sm))` (to refer to `CFLAGS32`/`CFLAGS64`), we must set `arch-bits-$(sm)` for each TA name.

External TAs use _ta_dev_kit.mk_, where `sm` is `ta_arm32` or `ta_arm64`, so a different path.

Reply:
Aha, thanks for the clarification. It might be worth adding a comment like:
_# $(sm) here is the name of the binary being built, for instance "avb" or "pkcs11"_